service/matching/tasklist/adaptive_scaler.go

```diff
@@ -291,9 +291,9 @@ func (a *adaptiveScalerImpl) adjustReadPartitions(m *aggregatePartitionMetrics,
 				a.logger.Warn("failed to get partition metrics", tag.WorkflowTaskListName(a.taskListID.GetPartition(i)), tag.Error(err))
 				break
 			}
-			p = toPartitionMetrics(i, resp)
+			p = a.toPartitionMetrics(i, resp)
 		}
-		if p.readOnly && p.backlog == 0 {
+		if p.readOnly && p.empty {
```

Comment:
I guess we don't have an explanation for the in-memory counter deviating, should we log this?
```
if backlog == 0 && p.empty {
    // log discrepancy with warn maybe? 
}```

Reply:
Empty was just introduced as a field, so this would currently have a lot of false positives.